pandas/core/arrays/period.py

```diff
@@ -487,7 +471,7 @@ def take(self, indices, allow_fill=False, fill_value=None):
         return self._from_ordinals(new_values, self.freq)
 
     def isna(self):
-        return self._data == iNaT
+        return self._isnan
```

Comment:
@jbrockmendel I don't think we should use `self._isnan` here. This is a concept from `Index`, that we should not need to copy for the arrays (we will need to move back `_isnan` to the Index anyway I think, as there are some Index internal methods that make use of it)

Reply:
The topic of whether _isnan belongs in the EA subclasses I think should be discussed separately/later.  As long as it does exist, it should be used instead of duplicated.